### ISSUE TITLE
dartdoc failed: Stack Overflow

### ISSUE THREAD

**Issue by sigurdm** [dart-lang/dartdoc#2104]
Seen on pub.dev

Seems different from: https://github.com/dart-lang/dartdoc/issues/2103

Package:
https://storage.googleapis.com/pub-packages/packages/over_react_test-2.8.0.tar.gz

Stack trace:
```
dartdoc failed: Stack Overflow
package:analyzer/src/dart/element/type.dart 1088:3                InterfaceTypeImpl.isDartCoreObject
package:analyzer/src/generated/type_system.dart 1131:16           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1224:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1217:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1256:13           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1224:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1217:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1762:16           Dart2TypeSystem._isInterfaceSubtypeOf
package:analyzer/src/generated/type_system.dart 1323:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1302:11           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1224:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1217:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1256:13           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1224:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1217:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1762:16           Dart2TypeSystem._isInterfaceSubtypeOf
package:analyzer/src/generated/type_system.dart 1323:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1302:11           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1224:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1217:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1256:13           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1224:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1217:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1762:16           Dart2TypeSystem._isInterfaceSubtypeOf
package:analyzer/src/generated/type_system.dart 1323:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1302:11           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1224:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1217:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1256:13           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1224:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1217:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1762:16           Dart2TypeSystem._isInterfaceSubtypeOf
package:analyzer/src/generated/type_system.dart 1323:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1302:11           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1224:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1217:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1256:13           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1224:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1217:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1762:16           Dart2TypeSystem._isInterfaceSubtypeOf
package:analyzer/src/generated/type_system.dart 1323:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1302:11           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1224:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1217:14           Dart2TypeSystem.isSubtypeOf
.                                                                 ...
.                                                                 ...
package:analyzer/src/generated/type_system.dart 1224:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/generated/type_system.dart 1217:14           Dart2TypeSystem.isSubtypeOf
package:analyzer/src/task/strong/checker.dart 1610:20             _OverrideChecker._findCovariantChecksForMember.<fn>
package:analyzer/src/dart/element/type.dart 791:20                FunctionTypeImpl.relateTypeFormals2
package:analyzer/src/task/strong/checker.dart 1606:34             _OverrideChecker._findCovariantChecksForMember
package:analyzer/src/task/strong/checker.dart 1538:9              _OverrideChecker._findCovariantChecks
package:analyzer/src/task/strong/checker.dart 1484:18             _OverrideChecker._checkForCovariantGenerics
package:analyzer/src/task/strong/checker.dart 1420:5              _OverrideChecker.check
package:analyzer/src/task/strong/checker.dart 330:22              CodeChecker.visitClassDeclaration
package:analyzer/src/dart/ast/ast.dart 1522:49                    ClassDeclarationImpl.accept
package:analyzer/src/dart/ast/ast.dart 7511:20                    CompilationUnitImpl.visitChildren
package:analyzer/src/task/strong/checker.dart 351:10              CodeChecker.visitCompilationUnit
package:analyzer/src/dart/analysis/library_analyzer.dart 380:13   LibraryAnalyzer._computeVerifyErrors
package:analyzer/src/dart/analysis/library_analyzer.dart 166:9    LibraryAnalyzer.analyzeSync.<fn>.<fn>
dart:collection                                                   _LinkedHashMapMixin.forEach
package:analyzer/src/dart/analysis/library_analyzer.dart 165:13   LibraryAnalyzer.analyzeSync.<fn>
package:analyzer/src/generated/utilities_general.dart 258:15      _PerformanceTagImpl.makeCurrentWhile
package:analyzer/src/dart/analysis/library_analyzer.dart 164:34   LibraryAnalyzer.analyzeSync
package:analyzer/src/dart/analysis/library_analyzer.dart 115:14   LibraryAnalyzer.analyze.<fn>
package:analyzer/src/generated/utilities_general.dart 258:15      _PerformanceTagImpl.makeCurrentWhile
package:analyzer/src/dart/analysis/library_analyzer.dart 114:43   LibraryAnalyzer.analyze
package:analyzer/src/dart/analysis/driver.dart 1342:65            AnalysisDriver._computeResolvedLibrary.<fn>
package:analyzer/src/dart/analysis/performance_logger.dart 34:15  PerformanceLog.run
package:analyzer/src/dart/analysis/driver.dart 1327:20            AnalysisDriver._computeResolvedLibrary
package:analyzer/src/dart/analysis/driver.dart 958:22             AnalysisDriver.performWork

```

**Comment by jcollins-g** [comment]
this looks very similar to dart-lang/sdk#39937.

**Comment by jcollins-g** [comment]
It's not quite the same, though.  At head SDK, we still crash but in a different way.

**Comment by jcollins-g** [comment]
```PACKAGE_NAME=over_react PACKAGE_VERSION=3.1.7 grind serve-pub-package```
 is sufficient to reproduce this this.  over_react_test also fails, but because it depends on over_react.

**Comment by jcollins-g** [comment]
A very small snippet of dart is sufficient to reproduce the crash.

```dart
void tryThis({void Function(bool isWorking) onMaybeWorking}) {
  onMaybeWorking?.call(true);
}
```

**Comment by jcollins-g** [comment]
fix mailed out for dart-lang/sdk#39996.   Once this lands and is published, we should be able to close this.

**Comment by jcollins-g** [comment]
fix published in analyzer 0.39.4.
